schedule/crontab.py

```diff
@@ -0,0 +1,649 @@
+        - A string representing a Crontab pattern.  This may get pretty
+          advanced, such as ``day_of_month='2-30/2'`` (for every even
+          numbered day) or ``day_of_month='1-7,15-21'`` (for the first and
+          third weeks of the month).
+
+    .. attribute:: month_of_year
+
+        - A (list of) integers from 1-12 that represents the months of
+          the year during which execution can occur.
+        - A string representing a Crontab pattern.  This may get pretty
+          advanced, such as ``month_of_year='*/3'`` (for the first month
+          of every quarter) or ``month_of_year='2-12/2'`` (for every even
+          numbered month).
+
+
+    It's important to realize that any day on which execution should
+    occur must be represented by entries in all three of the day and
+    month attributes.  For example, if ``day_of_week`` is 0 and
+    ``day_of_month`` is every seventh day, only months that begin
+    on Sunday and are also in the ``month_of_year`` attribute will have
+    execution events.  Or, ``day_of_week`` is 1 and ``day_of_month``
+    is '1-7,15-21' means every first and third Monday of every month
+    present in ``month_of_year``.
+    """
+
+    def __init__(
+        self,
+        minute="*",
+        hour="*",
+        day_of_week="*",
+        day_of_month="*",
+        month_of_year="*",
+        tz: Optional[str] = None,
+        **kwargs,
+    ):
+        self._orig_minute = cronfield(minute)
+        self._orig_hour = cronfield(hour)
+        self._orig_day_of_week = cronfield(day_of_week)
+        self._orig_day_of_month = cronfield(day_of_month)
+        self._orig_month_of_year = cronfield(month_of_year)
+        self._orig_kwargs = kwargs
+        self.hour = self._expand_cronspec(hour, 24)
+        self.minute = self._expand_cronspec(minute, 60)
+        self.day_of_week = self._expand_cronspec(day_of_week, 7)
+        self.day_of_month = self._expand_cronspec(day_of_month, 31, 1)
+        self.month_of_year = self._expand_cronspec(month_of_year, 12, 1)
+        self.tz = None
+        if tz is not None:
+            import pytz
+            if isinstance(tz, str):
+                self.tz = pytz.timezone(tz)  # type: ignore
+            elif isinstance(tz, pytz.BaseTzInfo):
+                self.tz = tz
+            else:
+                raise ValueError(
+                    "Timezone must be string or pytz.timezone object"
+                )
+
+    @classmethod
+    def from_expression(
+        cls, crontab_expression: str, tz: Optional[str] = None
+    ) -> "Crontab":
+        items = crontab_expression.split(" ")
+        if len(items) != 5:
+            raise ValueError(
+                "Invalid number of components in crontab expression"
+            )
+
+        return cls(
+            minute=items[0],
+            hour=items[1],
+            day_of_week=items[2],
+            day_of_month=items[3],
+            month_of_year=items[4],
```

Comment:
Isn't day of the week the 5th field instead of the third?
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07
https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#schedule